client/services/FeedbackGeneratorService.js

```diff
@@ -129,13 +132,15 @@ tie.factory('FeedbackGeneratorService', [
      *    name for the function needed to process/format the user's output.
      * @param {CorrectnessTest} correctnessTest
      * @param {*} observedOutput Actual output for running user's code.
+     * @param {Feedback} oldFeedback Feedback object to append feedback to. If
```

Comment:
Instead of adding this everywhere, would it make sense to add it in the caller instead? It keeps the contract of the individual functions clearer, and perhaps you can add a helper method to the Feedback object class along the lines of feedback.appendFeedback(newFeedback).